public/app/plugins/datasource/tempo/SearchTraceQLEditor/GroupByField.tsx

```diff
@@ -0,0 +1,141 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { AccessoryButton } from '@grafana/experimental';
+import { HorizontalGroup, Select, useStyles2 } from '@grafana/ui';
+import { notifyApp } from 'app/core/actions';
+import { createErrorNotification } from 'app/core/copy/appNotification';
+import { dispatch } from 'app/store/store';
+
+import { TraceqlFilter, TraceqlSearchScope } from '../dataquery.gen';
+import { TempoDatasource } from '../datasource';
+import { TempoQuery } from '../types';
+
+import InlineSearchField from './InlineSearchField';
+import { replaceAt } from './utils';
+
+interface Props {
+  datasource: TempoDatasource;
+  onChange: (value: TempoQuery) => void;
+  query: Partial<TempoQuery> & TempoQuery;
+}
+
+export const GroupByField = (props: Props) => {
+  const { datasource, onChange, query } = props;
+  const styles = useStyles2(getStyles);
+  const generateId = () => uuidv4().slice(0, 8);
+  const [isTagsLoading, setIsTagsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchTags = async () => {
+      try {
+        await datasource.languageProvider.start();
+        setIsTagsLoading(false);
+      } catch (error) {
+        if (error instanceof Error) {
+          dispatch(notifyApp(createErrorNotification('Error', error)));
+        }
+      }
+    };
+    fetchTags();
+  }, [datasource]);
+
+  useEffect(() => {
+    if (!query.groupBy || query.groupBy.length === 0) {
+      onChange({
+        ...query,
+        groupBy: [
+          {
+            id: generateId(),
+            scope: TraceqlSearchScope.Span,
+          },
+        ],
+      });
+    }
+  }, [onChange, query]);
+
+  const getTags = (f: TraceqlFilter) => {
+    return datasource!.languageProvider.getMetricsSummaryTags(f.scope);
+  };
+
+  const addFilter = () => {
+    updateFilter({
+      id: generateId(),
+      scope: TraceqlSearchScope.Span,
+    });
+  };
+
+  const removeFilter = (filter: TraceqlFilter) => {
+    const copy = { ...query };
+    onChange({ ...copy, groupBy: copy.groupBy?.filter((f) => f.id !== filter.id) });
```

Comment:
I think this is doing a copy 2 times doesn't it? Seems like what we pass to `onChange` should be enough to make this function pure.